internal/content_coding_test.go

```diff
@@ -113,3 +156,168 @@ func TestStreamGzipDecode(t *testing.T) {
 
 	require.Equal(t, []byte("howdy"), b[:n])
 }
+
+func BenchmarkZstdEncodeDecode(b *testing.B) {
+	enc, err := NewZstdEncoder(1)
+	require.NoError(b, err)
+	dec, err := NewZstdDecoder()
+	require.NoError(b, err)
+
+	for n := 0; n < b.N; n++ {
+		payload, err := enc.Encode([]byte("howdy"))
+		require.NoError(b, err)
+
+		actual, err := dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "howdy", string(actual))
+	}
+}
+
+func BenchmarkZstdReuse(b *testing.B) {
+	enc, err := NewZstdEncoder(1)
+	require.NoError(b, err)
+	dec, err := NewZstdDecoder()
+	require.NoError(b, err)
+
+	for n := 0; n < b.N; n++ {
+		payload, err := enc.Encode([]byte("howdy"))
+		require.NoError(b, err)
+
+		actual, err := dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "howdy", string(actual))
+
+		payload, err = enc.Encode([]byte("doody"))
+		require.NoError(b, err)
+
+		actual, err = dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "doody", string(actual))
+	}
+}
+
+func BenchmarkGzipEncodeDecode(b *testing.B) {
+	enc, err := NewGzipEncoder(-1)
+	require.NoError(b, err)
+	dec := NewGzipDecoder()
+
+	for n := 0; n < b.N; n++ {
+		payload, err := enc.Encode([]byte("howdy"))
+		require.NoError(b, err)
+
+		actual, err := dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "howdy", string(actual))
+	}
+}
+
+func BenchmarkGzipReuse(b *testing.B) {
+	enc, err := NewGzipEncoder(-1)
+	require.NoError(b, err)
+	dec := NewGzipDecoder()
+
+	for n := 0; n < b.N; n++ {
+		payload, err := enc.Encode([]byte("howdy"))
+		require.NoError(b, err)
+
+		actual, err := dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "howdy", string(actual))
+
+		payload, err = enc.Encode([]byte("doody"))
+		require.NoError(b, err)
+
+		actual, err = dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "doody", string(actual))
+	}
+}
+
+func BenchmarkZlibEncodeDecode(b *testing.B) {
+	enc, err := NewZlibEncoder(-1)
+	require.NoError(b, err)
+	dec := NewZlibDecoder()
+
+	for n := 0; n < b.N; n++ {
+		payload, err := enc.Encode([]byte("howdy"))
+		require.NoError(b, err)
+
+		actual, err := dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "howdy", string(actual))
+	}
+}
+
+func BenchmarkZlibEncodeDecodeWithTooLargeMessage(b *testing.B) {
+	enc, err := NewZlibEncoder(-1)
+	require.NoError(b, err)
+	dec := NewZlibDecoder()
+
+	for n := 0; n < b.N; n++ {
+		payload, err := enc.Encode([]byte("howdy"))
+		require.NoError(b, err)
+
+		_, err = dec.Decode(payload, 3)
+		require.ErrorContains(b, err, "size of decoded data exceeds allowed size 3")
+	}
+}
```

Comment:
This should be a test instead.

Reply:
This already exists as a test. I'll just remove it.